src/run.ts

```diff
@@ -186,6 +190,82 @@ async function destroy_stage(options: {
     });
   }
 
+  const stacks = await getAllStacksForStage(
+    `${process.env.REGION_A}`,
+    options.stage,
+    filters
+  );
+
+  const protectedStacks = stacks
+    .filter((i) => i.EnableTerminationProtection)
+    .map((i) => i.StackName);
+
+  if (protectedStacks.length > 0) {
+    console.log(
+      `We cannot proceed with the destroy because the following stacks have termination protection enabled:\n${protectedStacks.join(
+        "\n"
+      )}`
+    );
+  } else {
+    console.log(
+      "No stacks have termination protection enabled. Proceeding with the destroy."
+    );
+  }
+
+  const templates = await getCloudFormationTemplatesForStage(
+    `${process.env.REGION_A}`,
+    options.stage,
+    filters
+  );
+
+  const resourcesToCheck = [
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "FormAnswersTable",
+    },
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "FormQuestionsTable",
+    },
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "FormTemplatesTable",
+    },
+    { templateKey: `database-${options.stage}`, resourceKey: "FormsTable" },
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "StateFormsTable",
+    },
+    { templateKey: `database-${options.stage}`, resourceKey: "StatesTable" },
+    { templateKey: `database-${options.stage}`, resourceKey: "AuthUserTable" },
+    {
+      templateKey: `ui-${options.stage}`,
+      resourceKey: "CloudFrontDistribution",
+    },
+    { templateKey: `ui-auth-${options.stage}`, resourceKey: "CognitoUserPool" },
+  ];
+
+  const notRetained = resourcesToCheck.filter(
+    ({ templateKey, resourceKey }) => {
+      const policy =
+        templates?.[templateKey]?.Resources?.[resourceKey]?.DeletionPolicy;
+      return policy !== "Retain";
+    }
+  );
+
+  if (
+    ["master", "val", "production"].includes(options.stage) &&
+    notRetained.length > 0
```

Comment:
```suggestion
  if (
    ["master", "val", "production"].includes(options.stage) {
    # all the code that gets notRetained
      if (notRetained.length > 0) {
       # then the code to bail
```
This would make it so we only make the calls to check all this stuff if we are talking about an isDev stage.